core/rewards.lua

```diff
@@ -412,6 +412,48 @@ function Quest:GetStatus()
     end
 end
 
+-------------------------------------------------------------------------------
+------------------------------------ RECIPE -----------------------------------
+-------------------------------------------------------------------------------
+
+local Recipe = Class('Recipe', Item, {
+    display_option = 'show_recipe_rewards',
+    type = L['recipe']
+})
+
+function Recipe:Initialize(attrs)
+    Item.Initialize(self, attrs)
+
+    if not self.profession then
+        error('Recipe() reward requires a profession id to be set')
+    end
+end
+
+-- Tooltip Documentation:
+-- https://wowpedia.fandom.com/wiki/Patch_10.0.2/API_changes
+function Recipe:IsObtained()
+    local info = C_TooltipInfo.GetItemByID(self.item)
+    if info then
+        TooltipUtil.SurfaceArgs(info)
+        for _, line in ipairs(info.lines) do
+            TooltipUtil.SurfaceArgs(line)
+            if line.leftText and line.leftText == _G.ITEM_SPELL_KNOWN then
+                return true
+            end
+        end
+    end
+    return false
+end
```

Comment:
This implementation seems to work better than the previous `IsSpellKnown()` one. It correctly shows my learned recipes from the primal storms and tuskarr feast. After we merge and release this, we may want to consider moving this functionality to a utility function and adding an in-memory cache so it does not need to call these functions every time the tooltip is displayed or refreshed. We would need to listen for the event that fires when a recipe is learned, and then clear that item from the cache so it updates from false => true correctly.